From git@z Thu Jan  1 00:00:00 1970
Subject: [PATCH RFC] arm64: export this_cpu_has_cap
From: Arnd Bergmann <arnd@kernel.org>
Date: Fri, 29 Oct 2021 13:30:16 +0200
Message-Id: <20211029113023.760421-1-arnd@kernel.org>
To: Catalin Marinas <catalin.marinas@arm.com>, Will Deacon <will@kernel.org>, Mathieu Poirier <mathieu.poirier@linaro.org>, Suzuki K Poulose <suzuki.poulose@arm.com>, Anshuman Khandual <anshuman.khandual@arm.com>
Cc: Arnd Bergmann <arnd@arndb.de>, Marc Zyngier <maz@kernel.org>, Mark Rutland <mark.rutland@arm.com>, David Brazdil <dbrazdil@google.com>, linux-arm-kernel@lists.infradead.org, linux-kernel@vger.kernel.org
List-Id: <linux-kernel.vger.kernel.org>
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 7bit

From: Arnd Bergmann <arnd@arndb.de>

It's now used in a coresight driver that can be a loadable module:

ERROR: modpost: "this_cpu_has_cap" [drivers/hwtracing/coresight/coresight-trbe.ko] undefined!

Fixes: 8a1065127d95 ("coresight: trbe: Add infrastructure for Errata handling")
Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Acked-by: Will Deacon <will@kernel.org>
Acked-by: Catalin Marinas <catalin.marinas@arm.com>
Reviewed-by: Suzuki K Poulose <suzuki.poulose@arm.com>
Tested-by: Suzuki K Poulose <suzuki.poulose@arm.com>
Link: https://lore.kernel.org/r/20211029113023.760421-1-arnd@kernel.org
---
Not sure if we actually want this to be exported, this is my local
workaround for the randconfig build bot.
---
 arch/arm64/kernel/cpufeature.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/kernel/cpufeature.c b/arch/arm64/kernel/cpufeature.c
index ecbdff795f5e..beccbcfa7391 100644
--- a/arch/arm64/kernel/cpufeature.c
+++ b/arch/arm64/kernel/cpufeature.c
@@ -2864,6 +2864,7 @@ bool this_cpu_has_cap(unsigned int n)
 
 	return false;
 }
+EXPORT_SYMBOL(this_cpu_has_cap);
 
 /*
  * This helper function is used in a narrow window when,

-- 
2.29.2


